### ISSUE TITLE
ServerConnector.shutdown?

### ISSUE THREAD

**Issue by zhkl0228** [ptrd/kwik#44]
After a ServerConnector instance is started, how to shut down the instance?

**Comment by ptrd** [comment]
Agree we need a shutdown method on ServerConnector.

**Comment by Mechite** [comment]
To add, would be nice if it implemented `AutoCloseable`

In my case, helps integration as a bean with DI framework, maybe its nice to be able to close a `ServerConnector` with try-with-resources as follows:

```java
try (ServerConnector connector = ServerConnector.builder()
        ...
        .build()) {
    connector.registerApplicationProtocol(...);
    connector.start();
    
    // perform operation that interfaces with connector.
    // when scope is exit, server is gone
}
```

and implementation is just to make shutdown method as `void close() throws Exception;`, and have `ServerConnector` extend `AutoCloseable`.

**Comment by ptrd** [comment]
Implemented in release 0.9.1
